code/modules/experisci/experiment/types/scanning.dm

```diff
@@ -79,7 +79,7 @@
 /datum/experiment/scanning/perform_experiment_actions(datum/component/experiment_handler/experiment_handler, atom/target)
 	var/contributing_index_value = get_contributing_index(target)
 	if (contributing_index_value)
-		scanned[contributing_index_value] += traits & EXPERIMENT_TRAIT_DESTRUCTIVE ? 1 : target
+		scanned[contributing_index_value] += traits & EXPERIMENT_TRAIT_DESTRUCTIVE ? 1 : WEAKREF(target)
```

Comment:
Once again, if you do not resolve() then you should consider just using `REF`. The downside is that the ref may not be unique (except in mobs, where it is guaranteed to be unique), but if that's not a critical issue then we can save memory.

Reply:
That's kinda why I'm doing it honestly, I want to prevent duplicates or errors. I guess I could resolve them at some point and if it returns null clear the list?